### ISSUE TITLE
[bug] Unable to use "specific names" as values for conf entries

### ISSUE THREAD

**Issue by mhabibav** [conan-io/conan#15777]
### Environment details

* Operating System+version: Irrelevant
* Compiler+version: Irrelevant
* Conan version: 2.0.17 (packaged as exe with the pyinstaller script)
* Python version: 3.12

### Steps to reproduce

1. In any recipe, setting a conf value to any of the following values results in an error: [copy, re, fnmatch, OrderedDict]. This is probably because there are modules / types with the same name imported in the file where `eval` is run. That results in the value being interpreted as something other than a string. 

Dummy scenario:
conanfile.py
```python3
from conan import ConanFile


class dummy(ConanFile):
    name = "fixed_name"
    version = "1.2.3"

    def build(self):
        x = self.conf.get("user.company.conf:conf")
        self.output.warning("|" * 100)
        self.output.warning(type(x))
        self.output.warning(x)
        self.output.warning("|" * 100)
```

command:
```bash
conan build . -c "user.company.conf:conf=fnmatch"
```

### Logs

```
PS C:\E\ws\temp\New folder (3)> conan build . -c "user.company.conf:conf=re"
ERROR: It's not possible to compose module values and str ones.
PS C:\E\ws\temp\New folder (3)> conan build . -c "user.company.conf:conf=OrderedDict"
ERROR: It's not possible to compose type values and str ones.
PS C:\E\ws\temp\New folder (3)> conan build . -c "user.company.conf:conf=fnmatch"
ERROR: It's not possible to compose module values and str ones.
```

**Comment by mhabibav** [comment]
The suspected file where this issue is present is probably: `conans/model/conf.py` 
Particularly the function `_get_evaluated_value(__v)` in line: `643` (or line `653` in the current version)

**Comment by memsharded** [comment]
Thanks very much for your report @mhabibav 

This is indeed a gap in the evaluation of confs, it didn't take into account the possibility of a match with an existing Python module. I am submitting a fix in https://github.com/conan-io/conan/pull/15779 for next 2.2 release.

**Comment by mhabibav** [comment]
Thanks for the quick fix.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15779 merged, it will be in next 2.2
